src/nupic/ntypes/ArrayBase.cpp

```diff
@@ -148,11 +146,21 @@ void ArrayBase::zeroBuffer() {
  * This allows external buffers to be carried in the Array structure.
  */
 void ArrayBase::setBuffer(void *buffer, size_t count) {
-  buffer_ = std::shared_ptr<char>((char *)buffer, nonDeleter());
+  NTA_CHECK(type_ != NTA_BasicType_SDR);
   count_ = count;
   capacity_ = count * BasicType::getSize(type_);
+  buffer_ = std::shared_ptr<char>((char *)buffer, nonDeleter());
   own_ = false;
 }
+void ArrayBase::setBuffer(SDR &sdr) {
+  type_ = NTA_BasicType_SDR;
+  buffer_ = std::shared_ptr<char>((char *)&sdr, nonDeleter());
+  count_ = sdr.size;
```

Comment:
you can call `setBuffer(std::shared_ptr<char>((char *)&sdr, nonDeleter()), sdr.size)`